CompilerSource/settings-parse/crawler.cpp

```diff
@@ -105,29 +84,29 @@ namespace extensions
   {
     all_platforms.clear();
     cout << "\n\n\n\nStarting platform inspection\n";
-    for (string ef = file_find_first(enigma_root + "ENIGMAsystem/SHELL/Platforms/*",fa_sysfile | fa_readonly | fa_directory | fa_nofiles); ef != ""; ef = file_find_next())
+    for (auto& ef : std::filesystem::directory_iterator(enigma_root/"ENIGMAsystem/SHELL/Platforms/"))
     {
-      const string ef_path = enigma_root + "ENIGMAsystem/SHELL/Platforms/" + ef + "/Info/About.ey";
+      if (!ef.is_directory()) continue;
+      const std::filesystem::path ef_path = enigma_root/"ENIGMAsystem/SHELL/Platforms"/ef/"Info/About.ey";
       ifstream ext(ef_path.c_str(), ios_base::in);
       if (!ext.is_open()) continue;
 
       cout << " - " << ef_path << ": Opened.\n";
-      ey_data dat = parse_eyaml(ext,ef);
+      ey_data dat = parse_eyaml(ext,ef.path().u8string());
       eyit hasname = dat.values.find("represents");
       if (hasname == dat.values.end()) {
-        cout << "Skipping invalid platform API under `" << ef << "': File does not specify an OS it represents.";
+        cout << "Skipping invalid platform API under `" << ef.path().u8string() << "': File does not specify an OS it represents.";
         continue;
       }
       
-      sdk_descriptor& sdk = all_platforms[toUpper(ef)];
+      sdk_descriptor& sdk = all_platforms[toUpper(ef.path().u8string())];
       sdk.name   = dat.get("name");
       sdk.author = dat.get("author");
       sdk.build_platforms = dat.get("build-platforms");
       sdk.description = dat.get("description");
       sdk.identifier  = dat.get("identifier");
       sdk.represents  = dat.get("represents");
     }
-    file_find_close();
```

Comment:
Why are we not finding files now?

Reply:
file_find is just iterator which i've replaced with fs::directory_iterator